tools/clang/unittests/SPIRV/FileTestFixture.cpp

```diff
@@ -0,0 +1,129 @@
+//===- FileTestFixture.cpp ------------- File Test Fixture Implementation -===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include <fstream>
+
+#include "FileTestFixture.h"
+#include "FileTestUtils.h"
+
+#include "effcee/effcee.h"
+
+namespace clang {
+namespace spirv {
+
+namespace {
+const char hlslStartLabel[] = "// Run:";
+const std::vector<std::string> checkCommands = {"CHECK:",       "CHECK-NEXT:",
+                                                "CHECK-SAME:",  "CHECK-DAG:",
+                                                "CHECK-LABEL:", "CHECK-NOT:"};
+
+/// \brief Checks if the given string is any of the valid effcee CHECK commands.
+bool isCheckCommand(std::string &s) {
+  return std::any_of(checkCommands.begin(), checkCommands.end(),
```

Comment:
We can make `checkCommands` as a static variable of this function. Making `checkCommands` as global and having another `checkCommands` as member field in the test fixture can cause unexpected variable hiding problems later.